tests/models/test_default_evaluator.py

```diff
@@ -499,3 +502,160 @@ def test_gen_multiclass_roc_curve():
+
+    for test_fn in (
+        non_str_metric_name,
+        non_numerical_metric_value,
+    ):
+        with pytest.raises(
+            MlflowException,
+            match=f"'{test_fn.__name__}' (.*) did not return metrics as a dictionary of "
+            "string metric names with numerical values",
+        ):
+            _evaluate_custom_metric(0, test_fn, eval_df, metrics)
+
+    def non_str_artifact_name(*_):
+        return {"a": 32.1, "b": 3}, {1: [1, 2, 3]}
+
+    with pytest.raises(
+        MlflowException,
+        match=f"'{non_str_artifact_name.__name__}' (.*) did not return artifacts as a "
+        "dictionary of string artifact names with their corresponding objects",
+    ):
+        _evaluate_custom_metric(0, non_str_artifact_name, eval_df, metrics)
+
+
+def test_evaluate_custom_metric_success():
+    eval_df = pd.DataFrame({"prediction": [1.2, 1.9, 3.2], "target": [1, 2, 3]})
+    metrics = _get_regressor_metrics(eval_df["target"], eval_df["prediction"])
+
+    def example_custom_metric(_, given_metrics):
+        return {
+            "example_count_times_1_point_5": given_metrics["example_count"] * 1.5,
+            "sum_on_label_minus_5": given_metrics["sum_on_label"] - 5,
+            "example_np_metric_1": np.float32(123.2),
+            "example_np_metric_2": np.ulonglong(10000000),
+        }
+
+    res_metrics, res_artifacts = _evaluate_custom_metric(0, example_custom_metric, eval_df, metrics)
+    assert res_metrics == {
+        "example_count_times_1_point_5": metrics["example_count"] * 1.5,
+        "sum_on_label_minus_5": metrics["sum_on_label"] - 5,
+        "example_np_metric_1": np.float32(123.2),
+        "example_np_metric_2": np.ulonglong(10000000),
+    }
+    assert res_artifacts is None
+
+    def example_custom_metric_with_artifacts(given_df, given_metrics):
+        return (
+            {
+                "example_count_times_1_point_5": given_metrics["example_count"] * 1.5,
+                "sum_on_label_minus_5": given_metrics["sum_on_label"] - 5,
+                "example_np_metric_1": np.float32(123.2),
+                "example_np_metric_2": np.ulonglong(10000000),
+            },
+            {
+                "pred_target_abs_diff": np.abs(given_df["prediction"] - given_df["target"]),
+                "example_dictionary_artifact": {"a": 1, "b": 2},
+            },
+        )
+
+    res_metrics_2, res_artifacts_2 = _evaluate_custom_metric(
+        0, example_custom_metric_with_artifacts, eval_df, metrics
+    )
+    assert res_metrics_2 == {
+        "example_count_times_1_point_5": metrics["example_count"] * 1.5,
+        "sum_on_label_minus_5": metrics["sum_on_label"] - 5,
+        "example_np_metric_1": np.float32(123.2),
+        "example_np_metric_2": np.ulonglong(10000000),
+    }
+    assert "pred_target_abs_diff" in res_artifacts_2 and res_artifacts_2[
+        "pred_target_abs_diff"
+    ].equals(np.abs(eval_df["prediction"] - eval_df["target"]))
```

Comment:
```suggestion
    assert "pred_target_abs_diff" in res_artifacts_2
    assert res_artifacts_2[
        "pred_target_abs_diff"
    ].equals(np.abs(eval_df["prediction"] - eval_df["target"]))
```

Can we split this assertion to simplify the assertion error message?

### Example

```python
def test_1():
    res_artifacts_2 = {"example_dictionary_artifact": {"a": 1, "b": 3}}
    assert "example_dictionary_artifact" in res_artifacts_2
    assert res_artifacts_2["example_dictionary_artifact"] == {"a": 1, "b": 2}


"""
# pytest error message

E       AssertionError: assert {'a': 1, 'b': 3} == {'a': 1, 'b': 2}
E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {'b': 3} != {'b': 2}
E         Full diff:
E         - {'a': 1, 'b': 2}
E         ?               ^
E         + {'a': 1, 'b': 3}...
E         
E         ...Full output truncated (2 lines hidden), use '-vv' to show
"""


def test_2():
    res_artifacts_2 = {"example_dictionary_artifact": {"a": 1, "b": 3}}
    assert "example_dictionary_artifact" in res_artifacts_2 and res_artifacts_2[
        "example_dictionary_artifact"
    ] == {"a": 1, "b": 2}


"""
# pytest error message

E       AssertionError: assert ('example_dictionary_artifact' in {'example_dictionary_artifact': {'a': 1, 'b': 3}} and {'a': 1, 'b': 3} == {'a': 1, 'b': 2}
E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {'b': 3} != {'b': 2}
E         Full diff:
E         - {'a': 1, 'b': 2}
E         ?               ^
E         + {'a': 1, 'b': 3}...
E         
E         ...Full output truncated (2 lines hidden), use '-vv' to show)
"""

```